frb_example/pure_dart/rust/src/api.rs

```diff
@@ -725,3 +725,111 @@ impl ConcatenateWith {
         Ok(())
     }
 }
+
+pub enum CustomError {
+    Error0(String),
```

Comment:
It seems common to have named fields in errors (even thiserror has such https://docs.rs/thiserror/1.0.31/thiserror/), e.g. `Error3 {some_field_name: String}`.

In addition, it is more natural to users, since named fields have nice names in its dart translation, while unnamed fields have names like "field0".

Thus, maybe we can add a bit of tests about named fields in such error tests?

e.g.

```rust
pub enum CustomError {
...
Error3 { found: String, expected: String }
}